addon/index.d.ts

```diff
@@ -0,0 +1,42 @@
+import { TemplateFactory } from 'htmlbars-inline-precompile';
```

Comment:
Do we need this import?

Reply:
I copy-pasted from `@ember-decorators/component`, but ended up pretty much removing and rewriting everything except for this import of course 😂